drivers/stepper/adi_tmc/tmc2130.c

```diff
@@ -0,0 +1,306 @@
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2025 Navimatix GmbH
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT adi_tmc2130
+
+#include "tmc2130_reg.h"
+#include "adi_tmc_spi.h"
+#include <zephyr/kernel.h>
+#include <zephyr/init.h>
+#include "zephyr/devicetree.h"
+#include "zephyr/device.h"
```

Comment:
```suggestion
#include <zephyr/device.h>
```

Nit: Could you organize the headers, right now you start with zephyr headers and then in between there are c std library headers and then again zephyr headers.

I would personally organize them in this way,

c headers

zephyr headers

driver specific headers

#include <zephyr/logging/log.h>
LOG_MODULE_REGISTER(tmc2130, CONFIG_STEPPER_LOG_LEVEL);